src/lando/main/models/repo.py

```diff
@@ -167,23 +158,27 @@ def save(self, *args, **kwargs):
         if not self.commit_flags:
             self.commit_flags = []
 
-        if self.scm is None:
-            if self._is_git_repo:
-                self.scm = self.GIT
-            elif self._is_hg_repo:
-                self.scm = self.HG
-            else:
-                raise ValueError("Could not determine repo type.")
+        if not self.scm:
+            self.scm = self._find_supporting_scm(self.pull_path)
 
         super().save(*args, **kwargs)
 
-    @property
-    def is_git_repo(self):
-        return self.scm is not None and self.scm == self.GIT
+    def _find_supporting_scm(self, pull_path) -> str:
+        """Loop through the supported SCM_IMPLEMENTATIONS and return a key representing
+        the first SCM claiming to support the given pull_path.
+
+        A fallback is in place for SCM_GIT, which is not yet part of the SCM_IMPLEMENTATIONS.
+        """
+        for scm, impl in SCM_IMPLEMENTATIONS.items():
+            if impl.repo_is_supported(pull_path):
+                return scm
+        if self._is_git_repo:
+            return SCM_GIT
+        raise ValueError(f"Could not determine repo type for {pull_path}")
 
     @property
-    def is_hg_repo(self):
-        return self.scm is not None and self.scm == self.HG
+    def is_git_repo(self):
```

Comment:
```suggestion
    def is_git_repo(self) -> bool:
```

Reply:
I won't fix this one, as this bit of code gets killed in #147.